### PR TITLE
Assignment5 MattJett-EWU & twofingerrightclick

### PR DESCRIPTION
**Pull request by twofingerrightclick**
@MattJett-EWU

### CODE REVIEW

**Comment by 7hawks** [comment]
I will review this.

File: SecretSanta/src/SecretSanta.Api/appsettings.Development.json [hunk 1]
```diff
@@ -1,9 +1,12 @@
 {
-  "Logging": {
-    "LogLevel": {
-      "Default": "Information",
-      "Microsoft": "Warning",
-      "Microsoft.Hosting.Lifetime": "Information"
+    "Logging": {
```
Comment (twofingerrightclick) [inline SecretSanta/src/SecretSanta.Api/appsettings.Development.json]: I need to read up on what this is doing

File: SecretSanta/src/SecretSanta.Api/Startup.cs [hunk 2]
```diff
@@ -14,20 +16,29 @@ namespace SecretSanta.Api
...
         // This method gets called by the runtime. Use this method to add services to the container.
         // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
-        public static void ConfigureServices(IServiceCollection services)
+        public void ConfigureServices(IServiceCollection services)
         {
-            services.AddMvc(options => options.EnableEndpointRouting = false);
+            services.AddDbContext<ApplicationDbContext>(options =>
+                options.EnableSensitiveDataLogging()
+                       .UseSqlite(_Configuration.GetConnectionString("DefaultConnection")));
+
```
Comment (twofingerrightclick) [inline SecretSanta/src/SecretSanta.Api/Startup.cs]: I think we have all the pieces here. Need to read up on this

File: SecretSanta/src/SecretSanta.Business/Dto/Group.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace SecretSanta.Business.Dto
+{
+    public class Group: GroupInput, IEntity
+    {
+        public int Id { get; set; }
+        public IList<UserGroup> UserGroups { get; } = new List<UserGroup>();
```
Comment (twofingerrightclick) [inline SecretSanta/src/SecretSanta.Business/Dto/Group.cs]: not sure how to work with this entity list of UserGroups. Like should it be returned on an api call?

File: SecretSanta/src/SecretSanta.Business/Dto/UserGroup.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+﻿namespace SecretSanta.Business.Dto
+{
+    public class UserGroup
```
Comment (twofingerrightclick) [inline SecretSanta/src/SecretSanta.Business/Dto/UserGroup.cs]: I don't think this class is necessary.

File: SecretSanta/src/SecretSanta.Web/SecretSanta.Web.csproj [hunk 5]
```diff
@@ -3,4 +3,8 @@
   <PropertyGroup>
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
```
Comment (twofingerrightclick) [inline SecretSanta/src/SecretSanta.Web/SecretSanta.Web.csproj]: Hey what are you doing here newtonSoft!!!

Comment (twofingerrightclick) [reply SecretSanta/src/SecretSanta.Web/SecretSanta.Web.csproj]: using microsoft json

File: .gitignore [hunk 6]
```diff
@@ -273,3 +273,8 @@ __pycache__/
 # tools/
 
 # End of https://www.gitignore.io/api/aspnetcore
+
+*.sqlite
+*.db
```
Comment (Keboo) [inline .gitignore]: 👍

File: SecretSanta/src/SecretSanta.Api/Startup.cs (hunk 2 above)
Comment (Keboo) [reply SecretSanta/src/SecretSanta.Api/Startup.cs]: Nicely done!

File: SecretSanta/src/SecretSanta.Business/Dto/Group.cs (hunk 3 above)
Comment (Keboo) [reply SecretSanta/src/SecretSanta.Business/Dto/Group.cs]: In general a nullable property with get and set is the "normal" way to do it

File: SecretSanta/test/SecretSanta.Api.Tests/Controllers/GiftControllTests.cs [hunk 7]
```diff
@@ -1,26 +1,388 @@
...
-        protected override Gift CreateEntity()
-            => new Gift(Guid.NewGuid().ToString(),
-                Guid.NewGuid().ToString(),
-                Guid.NewGuid().ToString(),
-                new User(Guid.NewGuid().ToString(), Guid.NewGuid().ToString()));
-    }
 
-    public class GiftInMemoryService : InMemoryEntityService<Gift>, IGiftService
-    {
+        private SecretSantaWebApplicationFactory Factory { get; set; } = null!;
```
Comment (Keboo) [inline SecretSanta/test/SecretSanta.Api.Tests/Controllers/GiftControllTests.cs]: `null!` values are unusual here. Better approach would be to disable the warning here.

**Pull request closed without merging**